File: Daily Leetcode/Day 54/sum_of_sqr_nbr.py



import math

class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        h = int(math.sqrt(c))
        for i in range(h, -1, -1):
            for j in range(i, -1, -1):
                # val = pow(i, 2) + pow(j, 2)
                # if val < c and (2 * pow(i, 2) < c):
                #     return False
                # elif val == c:
                #     return True
                print(f"{i} {j}")


s = Solution()
s.judgeSquareSum(4)

File: Daily Leetcode/old/Day 54/sum_of_sqr_nbr.c
// 633. Sum of Square Numbers
// Medium
// https://leetcode.com/problems/sum-of-square-numbers/

/* Given a non-negative integer c, decide whether there're two integers a and b such that a2 + b2 = c.
 */

#include "stdio.h"
#include "math.h"

bool judgeSquareSum(int c) {
    int h = (int) sqrt(c);
    for (long i = h; i >= 0; i -= 1) {
        double res = sqrt(c - i * i);
        long l_res = (long) res;
        if (l_res == res) return true;
    }
    return false;
}